cnc_ctrl_v1/GCode.cpp

```diff
@@ -745,11 +707,11 @@ void  G38(const String& readString) {
       float zgoto;
 
 
-      float currentZPos = zAxis.target();
+      float currentZPos = zAxis.read();
 
       zgoto      = sys.inchesToMMConversion * extractGcodeValue(readString, 'Z', currentZPos / sys.inchesToMMConversion);
       sys.feedrate   = sys.inchesToMMConversion * extractGcodeValue(readString, 'F', sys.feedrate / sys.inchesToMMConversion);
-      sys.feedrate = constrain(sys.feedrate, 1, MAXZROTMIN * abs(zAxis.getPitch()));
+      sys.feedrate = constrain(sys.feedrate, 1, sysSettings.maxZRPM * abs(zAxis.getPitch()));
```

Comment:
I really like the way you are using the sys structure. It feels similar to the way things are done on the GC side and it feels much more intuitive than what we had before